projects/RabbitMQ.Client/client/impl/AmqpVersion.cs

```diff
@@ -70,27 +72,36 @@ public AmqpVersion(int major, int minor)
         /// <summary>
         /// The AMQP specification major version number.
         /// </summary>
-        public int Major { get; private set; }
+        public int Major { get; }
 
         /// <summary>
         /// The AMQP specification minor version number.
         /// </summary>
-        public int Minor { get; private set; }
+        public int Minor { get; }
 
         /// <summary>
         /// Implement value-equality comparison.
         /// </summary>
         public override bool Equals(object other)
         {
-            return (other is AmqpVersion version) && (version.Major == Major) && (version.Minor == Minor);
+            return other is AmqpVersion version && Equals(version);
         }
 
+        public bool Equals(AmqpVersion other) => Major == other.Major && Minor == other.Minor;
+
+        public static bool operator ==(AmqpVersion left, AmqpVersion right) => left.Equals(right);
+
+        public static bool operator !=(AmqpVersion left, AmqpVersion right) => !left.Equals(right);
+
         /// <summary>
         /// Implement hashing as for value-equality.
         /// </summary>
         public override int GetHashCode()
         {
-            return 31*Major.GetHashCode() + Minor.GetHashCode();
+            unchecked
+            {
+                return (Major * 397) ^ Minor;
+            }
```

Comment:
```suggestion
            return (Major, Minor).GetHashCode();
```

Reply:
That seems to require an extra dependency on `System.Tuple_2`, though.

Reply:
Thank you for the suggestion, it is definitely a nicer option but I'm not sure if it's worth an extra dependency (if it does require one, I'm unable to compile this on .NET Core 3.1 on MacOS as is).

Reply:
ok

Reply:
Due to .NET 4.6.1 this syntax isn't yet possible to use without additional dependencies. (Similar to that we can't use anonymous (int, int) return values)